lib/vagrant-libvirt/action/create_domain.rb

```diff
@@ -167,6 +167,19 @@ def call(env)
 
             disk[:absolute_path] = storage_prefix + disk[:path]
 
+            if not disk[:pool].nil?
+              disk_pool_name = disk[:pool]
+              @logger.debug "Overriding pool name with: #{disk_pool_name}"
+              disk_storage_pool = env[:machine].provider.driver.connection.client.lookup_storage_pool_by_name(disk_pool_name)
+              raise Errors::NoStoragePool if disk_storage_pool.nil?
+              xml = Nokogiri::XML(disk_storage_pool.xml_desc)
+              disk_storage_prefix = xml.xpath('/pool/target/path').inner_text.to_s + '/'
```

Comment:
Is it worth trying to make this a function to reuse above as well for retrieving the storage pool? Though I am wondering what happens in the the case of `config.qemu_use_session` being set and why it's guarding the storage pool retrieval above. The commit making the change to guard that just mentions qemu_use_session bug fixes, so I'm not sure if there is a potential issue

Reply:
yes, that would make sense, for POC i was copy/pasting the needed parts. What file would you place that function in?